docs/architecture/bitstream_format.rst

```diff
@@ -1,28 +1,135 @@
+The pattern is inserted before the 32-bit synchronization word and is ignored by the configuration state machine.
+
+Configuration Packets
++++++++++++++++++++++
+
+All words that follow the synchronization word are interpreted by the configuration logic.
+Depending on the architecture the words are interpreted as 16 or 32 bit big-endian words and form the configuration packets.
+There are three types of packets identified by the header which can contain three major commands: NOP, READ, WRITE.
+While NOP is used for inserting required delays in the configuration sequence the most common are read and write operations.
+These supported packet types are:
+
+* Type 0 - these packets exist only when performing zero-fill between rows
+* Type 1 - used for read/writes of a number of words specified by the word count portion of the header which differs between architectures
+* Type 2 - this type of packets expands the word count field that Type 1 packets have.
+  For Series-7 FPGAs Type 2 expands to 27 bits by omitting the register address, but has to follow a Type 1 packet which carries the address information.
+  Spartan6's Type 2 packets still contain the frame address, however the configuration logic expects the header to be followed by two 16-bit words with the MSB in the first word.
+
+
+Configuration Registers
++++++++++++++++++++++++
+
+The addresses that are specified in the configuration packets are mapped to a set of registers that provide low-level control over the chip.
+Only some of them take part in the programming sequence whereas the rest controls various physical aspects of the configuration interface.
+Some of the key registers used during programming are:
+
+* IDCODE - Before writing to the configuration memory, a 32-bit device ID code must be written to this register. Reads from the register return the attached device's ID code.
+* CRC - When a packet is received by the device, it automatically updates an internal CRC calculation to include the contents of that packet. A write to the CRC register checks that the calculated CRC matches the expected value written to the register. This CRC check is only used to provide integrity checking of the packet stream, not the configuration memory contents, and are not required for programming. If you are modifying a bitstream, CRC writes can simply be removed instead of recalculating them.
+* Command - Most of the programming sequence is implemented as a state machine that is controlled via one-shot actions. Writes to this register arm an action that, depending on the action requested, may be triggered immediately or delayed until some other condition is met.
+Important Note: During autoincremented frame writes (described later), the current command is rewritten during every autoincrement. This has the effect of rearming the action on every frame written.
+* Frame Address Register (FAR) - Writes to this register set the starting address for the next frame read or write.
```

Comment:
Indent for FAR and FDRI is not correct.  Maybe needs another space or two?

Reply:
Fixed.